crates/sui-core/src/consensus_types/committee_api.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::consensus_types::AuthorityIndex;
+use narwhal_config::committee::AuthorityIdentifier;
+use sui_types::base_types::AuthorityName;
+use sui_types::committee::StakeUnit;
+
+pub(crate) trait CommitteeAPI {
+    fn total_stake(&self) -> StakeUnit;
+    fn authority_pubkey_by_index(&self, index: AuthorityIndex) -> Option<AuthorityName>;
+    fn authority_hostname_by_index(&self, index: AuthorityIndex) -> Option<&str>;
+    fn authority_stake_by_index(&self, index: AuthorityIndex) -> StakeUnit;
+}
+
+impl CommitteeAPI for narwhal_config::Committee {
+    fn total_stake(&self) -> StakeUnit {
+        narwhal_config::Committee::total_stake(self)
+    }
+
+    fn authority_pubkey_by_index(&self, index: AuthorityIndex) -> Option<AuthorityName> {
```

Comment:
We discussed offline but to clarify again, the assumption here is that the authorities are ordered by their public keys  in the SUI side and the AuthorityIndex is assigned then from [0..committee_size), right?

Reply:
The index here is basically what's inside the AuthorityIdentity in Narwhal. No assumption is actually made here.